src/systems/filecoin_blockchain/chainsync/chainsync.id

```diff
@@ -0,0 +1,95 @@
+import ipld "github.com/filecoin-project/specs/libraries/ipld"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+
+type ChainSync struct {
+    // LastTrustedCheckpoint is the most recent TrustedCheckpoint
+    // ChainSync is aware of. This is used as the starting point from
+    // which to synchronize the chain. (See block/checkpoints)
+    LastTrustedCheckpoint  block.TrustedCheckpoint
+
+    // VGraph is the ValidationGraph ChainSync uses to track all
+    // Blocks and decide which blocks to validate, when.
+    VGraph                 ValidationGraph
+}
+
+type ValidationGraph struct {
+    // ChainStore is the ipld Datastore through which to load and store
+    // all Chain datastructures (Blocks, Messages, StateTrees, etc.)
+    ChainStore       ipld.Store
+
+    // CacheChainStore is the ipld Datastore through which to load and store
+    // chain datastructures that may be temporary and may be garbage collected.
+    CacheChainStore  ipld.Store
+
+    // Graph is a datastructure that provides efficient access to all
+    // nodes needed for validation, and keeps both forward and back links.
+    // Graph graph.BlockGraph
+
+    // FinalityTipset contains the latest tipset declared to be final.
+    // This is what we use to check against.
```

Comment:
To check what against? 
A 1-sentence recapitulation of what "final" means would help here too.

Reply:
addressed